### PR TITLE
1.15 Buzzy Bee Update

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR adds the new things from the Minecraft 1.15 Buzzy Bee update.

- Added new BeeData
I haven't done much EntityData stuff, so if anyone would like to see some changes, please let me know. I'm not 100% sold on the patterns so I'm looking for feedback on them.

- Added new bee related particles

PS yes I know this errors during build, thats because we are still relying on Paper 1.14.4. No rush to merge this (it works on my tester version, because I added Spigot 1.15)

---
**Target Minecraft Versions:** 1.15+
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/entity/BeeData.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	@Override
+	protected boolean equals_i(EntityData<?> obj) {
+		if (!(obj instanceof BeeData))
+			return false;
+		final BeeData other = (BeeData) obj;
+		if (angry != other.angry)
+			return false;
+		if (nectar != other.nectar)
+			return false;
+		return true;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/entity/BeeData.java]: Can be simplified to
```
return (angry == other.angry) && (nectar == other.nectar)
```

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/entity/BeeData.java]: ah yes. I think I was changing 100 things (this went thru a bunch of revisions) and I forgot to revery this one.

File: src/main/java/ch/njol/skript/entity/BeeData.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			return false;
+		final BeeData other = (BeeData) obj;
+		return (angry == other.angry) && (nectar == other.nectar);
+	}
+	
+	@Override
+	public boolean isSupertypeOf(EntityData<?> e) {
+		if (e instanceof BeeData)
+			return (angry == 0 || ((BeeData) e).angry == angry) && (nectar == 0 || ((BeeData) e).nectar == nectar);
+		return false;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/entity/BeeData.java]: Can be simplified to
```
return (e instanceof BeeData) && (angry == 0 || ((BeeData) e).angry == angry) && (nectar == 0 || ((BeeData) e).nectar == nectar)
```

File: src/main/java/ch/njol/skript/entity/BeeData.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			EntityData.register(BeeData.class, "bee", Bee.class, 2,
+				"no nectar bee", "happy bee", "bee", "nectar bee", "angry bee", "angry nectar bee");
+		}
+	}
+	
+	private int nectar = 0;
+	private int angry = 0;
+	
+	@Override
+	protected boolean init(Literal<?>[] exprs, int matchedPattern, SkriptParser.ParseResult parseResult) {
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/entity/BeeData.java]: Should be using ``ParseResult`` instead of ``SkriptParser.ParseResult``

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/entity/BeeData.java]: done, good catch!

File: src/main/java/ch/njol/skript/entity/BeeData.java (hunk 2 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/entity/BeeData.java]: Done, definitely makes that simpler.

**Comment by ShaneBeee** [comment]
Closing this and starting fresh, had a little LF->CRLF issue

**Pull request closed without merging**